stdlib/public/core/ChunkedCollection.swift

```diff
@@ -0,0 +1,127 @@
+//===--- ChunkedCollection.swift ----------------------------------------------------===//
+//
+// This source file is part of the Swift.org open source project
+//
+// Copyright (c) 2014 - 2017 Apple Inc. and the Swift project authors
+// Licensed under Apache License v2.0 with Runtime Library Exception
+//
+// See https://swift.org/LICENSE.txt for license information
+// See https://swift.org/CONTRIBUTORS.txt for the list of Swift project authors
+//
+//===----------------------------------------------------------------------===//
+
+/// A collection that presents the elements of its base collection
+/// in `SubSequence` chunks of any given size.
+///
+/// A ChunkedCollection is a lazy view on the base Collection, but it does not implicitly confer
+/// laziness on algorithms applied to its result.  In other words, for ordinary collections `c`:
+///
+/// * `c.chunks(of: 3)` does not create new storage
+/// * `c.chunks(of: 3).map(f)` maps eagerly and returns a new array
+/// * `c.lazy.chunks(of: 3).map(f)` maps lazily and returns a `LazyMapCollection`
+@_fixed_layout
+public struct ChunkedCollection<Base: Collection> {
+    @usableFromInline
+    internal let _base: Base
+    @usableFromInline
+    internal let _size: Int
+    
+    ///  Creates a view instance that presents the elements of `base`
+    ///  in `SubSequence` chunks of the given size.
+    ///
+    /// - Complexity: O(1)
+    @inlinable
+    internal init(_base: Base, _size: Int) {
+      self._base = _base
+      self._size = _size
+    }
+}
+
+extension ChunkedCollection: Collection {
+  public struct Index {
+    @usableFromInline
+    let _base: Base.Index
+        
+    @usableFromInline
+    init(_base: Base.Index) {
+      self._base = _base
+    }
+  }
+    
+  public typealias Element = Base.SubSequence
+  public var startIndex: Index { return Index(_base: _base.startIndex) }
+  public var endIndex: Index { return Index(_base: _base.endIndex) }
+    
+  public subscript(i: Index) -> Element {
+    let range = i..<index(after: i)
+    return _base[range.lowerBound._base..<range.upperBound._base]
+  }
+    
+  @inlinable
+  public func index(after i: Index) -> Index {
+    return Index(_base: _base.index(i._base, offsetBy: _size, limitedBy: _base.endIndex) ?? _base.endIndex)
+  }
+}
+
+extension ChunkedCollection.Index: Comparable {
```

Comment:
As of the latest swift, this can be synthesized.

Reply:
@airspeedswift This one got me confused, can comparable conformance be synthesized? I thought it was only for enums.
The other comments were fixed 🎉